src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,278 @@
+
+If and when a new TypeScript version causes a type error in your code, consider trying each of the following strategies in order until it is fixed:
+
+1. Fix small issues with the types to work with the new TypeScript behavior
+2. Refactor the types to reduce complexity and work with the new TypeScript behavior
+3. Use `any` on types you can't figure out, with a `// TODO` comment linked to a tracking issue/ticket for cleaning it up later
+4. Use `// @ts-expect-error` on remaining issues you can't figure out, with a `// TODO` comment linked to a tracking issue/ticket for cleaning it up later
+
+Let's look at those tips in more details.
+
+#### Straightforward, Clean Types
+
+TypeScript's type system is fantastically powerful and can represent some wild and wacky type operations.
+However, the more complex your types are, the more difficult they are to read, write, and update over time.
+TypeScript types follow simplicity principles from other types of code:
+
+- Prefer simple, easy-to-read code (types) whenever possible
+- Prefer well-named types that explain what they're doing
+- When code does become complex to the point of being difficult to read, consider using comments to explain the tricky details
+
+Simpler types are less likely to be broken by new TypeScript versions tinkering with nuances of the type system.
+They're also easier to debug when something becomes broken.
+
+Therefore, if you're unable to fix small issues with newly broken types, you might benefit from trying to refactor them to be simplier and easier to work with.
+
+#### Falling Back to `any`
+
+Another general type system best practice is to avoid the `any` type.
+The `any` type indicates that a value that can be <em>any</em>thing and TypeScript should allow using it in almost any way.
+Which is as dangerous as it sounds: `any` stops TypeScript from reporting potentially useful type errors and stops other [TypeScript developer utilities](https://code.visualstudio.com/docs/languages/typescript#_code-navigation) from working as well.
+
+Still, if you're struggling to get a type to work, `any` can be an effective backup band-aid to lessen TypeScript's type checking strictness.
+Replacing a type with `any` can save you from having to dive into complexities around that type.
+Try to always add a `// TODO` comment explaining why you used the `any`, to help inform future efforts to remove the `any`.
+
+:::tip
+The [`@typescript-eslint/no-explicit-any`](https://typescript-eslint.io/rules/no-explicit-any) lint rule can enforce against explicitly writing `any` types in code.
+:::
+
+#### Falling Back to TypeScript Comment Directives
+
+Even more dangerous than `any` are the TypeScript _comment directives_ that completely turn off the type checker for a line:
+
+- `// @ts-ignore`: Silences any type checking errors for a line
+- `// @ts-expect-error`: Acts like `// @ts-ignore`, but if a type error isn't reported, causes a new type error for the unnecessary comment
```

Comment:
I was tripped up by this wording. After re-reading and getting more context from the following paragraphs, consider:
"Acts like ... but if the corresponding line doesn't actually throw an error, TS will show an error that this comment is necessary"